Integrations/integration-Tanium.yml

```diff
@@ -0,0 +1,483 @@
+        response = handler.ask(**kwargs)
+
+        query_text = response['question_object'].query_text
+        if response['question_results']:
+            export_kwargs = {}
+            export_kwargs['obj'] = response['question_results']
+            export_kwargs['export_format'] = 'csv'
+            out = handler.export_obj(**export_kwargs)
+
+            result = csvstr_to_list(out)
+
+            ec = { 'Tanium.QuestionResults' : result }
+            return create_entry(
+                    'Result for parsed query - %s' % (query_text,),
+                    result,
+                    ec)
+        else:
+            return 'Parsed query - %s\nNo results were found' % (query_text,)
+
+    def createPackage(handler):
+        kwargs = {}
+        for key, value in demisto.args().items():
+            kwargs[key] = value
+        if demisto.get(demisto.args(), 'file_urls'):
+            kwargs['file_urls'] = demisto.args()['file_urls'].split(",")
+        response = handler.create_package(**kwargs)
+        parsed = parseToJson(handler, response)
+        final_result = create_entry('Tanium Package', [parsed], {'Tanium.Packages' : parsed}, ['id', 'name', 'creation_time', 'command'])
+        return final_result
+
+    def restoreSoutAndExit(final_result):
+        sys.stdout = sout
+        demisto.results(final_result)
+        sys.exit(0)
+
+    # Dealing with Broken Pipe issues raised by some pytan commands
+    from cStringIO import StringIO
+    sout = sys.stdout
+    sys.stdout = StringIO()
+
+    handler = getHandler()
+    response = ''
+
+    if demisto.command() == 'test-module':
+        final_result = getAllPackages(handler)
+        restoreSoutAndExit('ok')
+    if demisto.command() == 'tn-get-package':
+        final_result = getPackage(handler)
+    if demisto.command() == 'tn-get-saved-question':
+        final_result = getSavedQuestion(handler)
+    if demisto.command() == 'tn-get-object':
+        final_result = getObject(handler, unicode(demisto.args()['object_type']))
+    if demisto.command() == 'tn-get-all-packages':
+        final_result = getAllPackages(handler)
+    if demisto.command() == 'tn-get-all-saved-questions':
+        final_result = getAllSavedQuestions(handler)
+    if demisto.command() == 'tn-deploy-package':
+        final_result = deployAction(handler)
+    if demisto.command() == 'tn-ask-system':
+        final_result = askQuestion(handler, 'Get Computer Name from all machines with Computer Name matching \"'+ demisto.args()['hostname'] + '\"')
+    if demisto.command() == 'tn-ask-question':
+        final_result = askQuestion(handler, demisto.args()['question'])
+    if demisto.command() == 'tn-create-package':
+        final_result = createPackage(handler)
+    if demisto.command() == 'tn-approve-saved-action':
+        final_result = approveSavedAction(handler)
+
+    restoreSoutAndExit(final_result)
+  type: python
+  commands:
+  - name: tn-get-package
+    arguments:
+    - name: name
```

Comment:
I think that one of name and id arguments should be required. and one of them should be default. No?

Reply:
you need to use either one. if I set one to required you won't be able to use it without it.